rich/progress.py

```diff
@@ -794,6 +1132,151 @@ def track(
                 advance(task_id, 1)
                 refresh()
 
+    def wrap_file(
+        self,
+        file: BinaryIO,
+        total: Optional[int] = None,
+        task_id: Optional[TaskID] = None,
+        description: str = "Reading...",
+    ) -> BinaryIO:
+        """Track progress file reading from a binary file.
+
+        Args:
+            file (BinaryIO): A file-like object opened in binary mode.
+            total (int, optional): Total number of bytes to read. This must be provided unless a task with a total is also given.
+            task_id (TaskID): Task to track. Default is new task.
+            description (str, optional): Description of task, if new task is created.
+
+        Returns:
+            BinaryIO: A readable file-like object in binary mode.
+
+        Raises:
+            ValueError: When no total value can be extracted from the arguments or the task.
+        """
+        # attempt to recover the total from the task
+        if total is None and task_id is not None:
+            with self._lock:
+                total = self._tasks[task_id].total # type: ignore
```

Comment:
The type ignore feels wrong here since we know the types of the parameter `total` and `Task.total`.

I suspect if you created a new variable e.g. `float_total` and assigned to that rather than re-assigning `total` to a float, you wouldn't need to ignore.

Reply:
I added a new variable that is an `Optional[float]`, so this should be fixed now.